tests/test_openmc_utils.py

```diff
@@ -0,0 +1,274 @@
+                           name='1001.71c', path='293.6K/H_001_293.6K.ace',
+                           temperature='2.53e-08', zaid='1001'),
+           openmc_utils.AceTable(alias='Am-242m.73c', awr='239.9801', location='1',
+                           metastable='1', name='95242.73c',
+                           path='900K/Am_242_900K.ace', temperature='7.756e-08',
+                           zaid='95242'),
+           openmc_utils.AceTable(awr='89.1324', location='1', name='ZrZrH.71t',
+                           path='tsl/zrzrh.acer', temperature='2.551e-08',
+                           zaid='0')]
+    assert_equal(exp, xs.ace_tables)
+
+
+def test_cross_sections_abspath_with_dir():
+    xs = openmc_utils.CrossSections(sample_xs_with_dir)
+    assert_equal('ascii', xs.filetype)
+    assert_equal(xs.path, "/")
+
+    exp_abspaths = ["/293.6K/H_001_293.6K.ace",
+                    "/900K/Am_242_900K.ace", "/tsl/zrzrh.acer"]
+    obs_abspaths = [table.abspath for table in xs.ace_tables]
+    assert_equal(exp_abspaths, obs_abspaths)
+
+
+def test_cross_sections_mcnp_id():
+    xstables = openmc_utils.CrossSections(sample_xs_with_mcnp_id).ace_tables
+    mcnp_obs = [table.nucid for table in xstables if table.alias == "Co-58m.70c"][0]
+    assert_equal(mcnp_obs, 270580001)
+    nucid_obs = [table.nucid for table in xstables if table.alias == "Co-58.70c"][0]
+    assert_equal(nucid_obs, 270580000)
+
+def test_cross_sections_roundtrip():
+    sample_xs.seek(0)
+    xs = openmc_utils.CrossSections(sample_xs)
+    sample_xs.seek(0)
+    exp = sample_xs.read()
+    obs = xs.xml()
+    assert_equal(exp, obs)
+
+def test_calc_structured_coords():
+    lower_left = np.array([0.0, 0.0, 0.0])
+    upper_right = np.array([1.0, 2.0, 3.0])
+    dimension = np.array([2, 4, 5])
+    exp_structured_coords = np.array([[0.0, 0.5, 1.0],
+                                      [0.0, 0.5, 1.0, 1.5, 2.0],
+                                      [0.0, 0.6, 1.2, 1.8, 2.4, 3.0]])
+    structured_coords = openmc_utils.calc_structured_coords(lower_left,
+            upper_right, dimension)
+    assert_equal(len(structured_coords), len(exp_structured_coords))
+    for i in range(len(exp_structured_coords)):
+        assert_array_almost_equal(structured_coords[i],
+                exp_structured_coords[i])
+
+def test_get_e_bounds_from_openmc_sp():
+    if not HAVE_PYMOAB or sys.version_info[0] == 2:
+        raise SkipTest
+    try:
+        import openmc
+    except:
+        raise SkipTest
+    # energy bin: [0.0, 1.0, 20.0], 2bins
+    # 6 volume elemenes
+    filename = os.path.join(cwd, "files_test_openmc", "statepoint.10.ebin2.ves6.h5")
+    # OpenMC energy unit is eV
+    exp_e_bounds = np.array([0.0, 1.0, 20.0]) * 1e6
+    e_bounds = openmc_utils.get_e_bounds_from_openmc_sp(filename, tally_id=1)
+    assert_array_equal(e_bounds, exp_e_bounds)
+
+def test_flux_changes_order():
+    if not HAVE_PYMOAB or sys.version_info[0] == 2:
```

Comment:
This test can be run if we don't have `openmc` or `pymoab`

Reply:
You are right. Updated in https://github.com/pyne/pyne/pull/1202/commits/88003563c7db8e812f5aa5749c7f09dcf9b63c2b.